meson.build

```diff
@@ -205,6 +205,9 @@ if get_option('enable-tizen')
   add_project_arguments('-D__TIZEN__=1', language: ['c', 'cpp'])
 endif
 
+if get_option('enable-test')
+  add_project_arguments('-D__UNITTEST__=1', language: ['c', 'cpp'])
```

Comment:
With the current logic, ```__UNITTEST__``` should be enabled ONLY WHEN the objects are built for gtest, but you are enabling it when the project has unit-test enabled.

You need to modify how gtest binaries are built, not the global build.